src/test/java/api/subscribers/LoanRelatedFeeFineClosedSubscriberApiTest.java

```diff
@@ -0,0 +1,141 @@
+package api.subscribers;
+
+import static api.support.matchers.ItemMatchers.isAvailable;
+import static api.support.matchers.ItemMatchers.isCheckedOut;
+import static api.support.matchers.ItemMatchers.isDeclaredLost;
+import static api.support.matchers.ItemMatchers.isLostAndPaid;
+import static api.support.matchers.JsonObjectMatcher.hasJsonPath;
+import static api.support.matchers.LoanMatchers.isClosed;
+import static api.support.matchers.LoanMatchers.isOpen;
+import static api.support.matchers.ValidationErrorMatchers.hasErrorWith;
+import static api.support.matchers.ValidationErrorMatchers.hasMessage;
+import static api.support.matchers.ValidationErrorMatchers.hasNullParameter;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.UUID;
+
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.folio.circulation.support.http.client.Response;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.DeclareItemLostRequestBuilder;
+
+public class LoanRelatedFeeFineClosedSubscriberApiTest extends APITests {
+  private IndividualResource loan;
+  private IndividualResource item;
+
+  @Before
+  public void createLoanAndDeclareItemLost() {
+    useLostItemPolicy(lostItemFeePoliciesFixture.chargeFee().getId());
+
+    item = itemsFixture.basedUponSmallAngryPlanet();
+    loan = checkOutFixture.checkOutByBarcode(item, usersFixture.jessica());
+
+    declareLostFixtures.declareItemLost(new DeclareItemLostRequestBuilder()
+      .forLoanId(loan.getId()));
+  }
+
+  @Test
+  public void canCloseLoanIfAllFeesClosed() {
```

Comment:
I think I prefer `should` (I think you were the first person to use that in mod-circulation) when describing side-effects to `can`

```suggestion
  public void shouldCloseLoanWhenAllFeesClosed() {
```

Reply:
This test demonstrates that the loan is closed when the fees are paid. Are there other kinds of closure of the fees that need to be tested?

Are fines ever issued for lost items?

Reply:
As far as I know a fee/fine can be closed in case:
* It is paid fully;
* It is transferred fully;
* It is waived fully;
* It is cancelled.

I can add tests for all these cases, but I think this should be covered in mod-feesfines (in a fact it does not check the closure reason,  but has a simple condition - `status == Closed` and `remainingAmount == 0.00`)

Reply:
> Are fines ever issued for lost items?

No, it depends on lost-item policy, processing fee won't be issued:
* If processing fee amount == 0.00;
* If "Do not charge processing fee" is set to `true`;

item fee won't be issued:
* If it is of `set cost` type and amount == 0.00;
* If it is of `actual cost` type - in this case no fee will be created automatically but a fee will have to be created manually.

If there is no neither processing fee/item fee AND item fee is of `set cost` type - then loan will be closed immediately.

Reply:
Apologies for the vague question, I thought that a fine was sometimes issued when a lost item is then found and returned?

Reply:
> I can add tests for all these cases, but I think this should be covered in mod-feesfines (in a fact it does not check the closure reason, but has a simple condition - `status == Closed and remainingAmount == 0.00)`

Ok, I wanted to confirm my understanding that the decision was about closure not full payment.

I agree about not adding tests for each variation. Maybe one variation could be useful to demonstrate that the decision is independent of it being paid? Could that be a unit-like test instead, that shows that it's only closure that is important not how the fee was closed.

Reply:
In this case an overdue fine can be issued depending on policy configuration. But loan can be closed in this case as well, only lost fees are considered.

Reply:
> In this case an overdue fine can be issued depending on policy configuration. But loan can be closed in this case as well, only lost fees are considered.

Is that described in the story? If so, is there a test demonstrating that an overdue fine does not stop a loan from being closed as part of this process?

Reply:
It is described in https://issues.folio.org/browse/CIRC-717: 
![image](https://user-images.githubusercontent.com/9094650/82664452-6b45e380-9c3a-11ea-9dd4-7b891460c65a.png)

Will add a test that are demonstrating that loan can be closed in case non lost fee present .

Reply:
@bohdan-suprun In what part of CIRC-717 is the following situation described, which scenario?

```
Given an open loan
And an unpaid item processing fee
And an unpaid set cost fee
And an unpaid overdue fine is not paid
When both the fees are paid
Then the loan should be closed despite the outstanding fine
```

Reply:
It is not possible case i think.

The overdue fine is possible only when the lost item is returned. But lost item fees/overdue fines are refunded/assigned at the very end of the flow, at that time the loan is already closed.

However, it is possible to have lost fees + a manually created fee. This case described in https://issues.folio.org/browse/CIRC-743: 

![image](https://user-images.githubusercontent.com/9094650/82666178-055b5b00-9c3e-11ea-961d-36c48d9ec6e8.png)

Reply:
Thank you for explaining that to me.

Reply:
That note really should be a scenario. Thank you for adding tests that demonstrate how other fees or fines are ignored.